src/main/java/core/basesyntax/service/date/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.date;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private Map<String, Integer> storage;
+
+    public ReportGeneratorImpl(Map<String, Integer> storage) {
+        this.storage = storage;
+    }
+
+    public ReportGeneratorImpl() {
+    }
+
+    @Override
+    public String getReport() {
+        return storage.entrySet()
+                .stream()
+                .map(entry -> entry.getKey() + "," + entry.getValue())
+                .collect(Collectors.joining("\n"));
```

Comment:
This method does not handle the situation when the storage is null. It would be better to check if the storage is not null before you start to use it.